SConstruct

```diff
@@ -322,6 +322,15 @@ opts.Add("rcflags", "Custom flags for Windows resource compiler")
 # in following code (especially platform and custom_modules).
 opts.Update(env)
 
+# Warn user about unrecognized or misspelled builtin_* options.
+# Currently it works only for command line variables
+# TODO: Add the same check for options defined inside profile
+# file. Look at alternative profile loading approach in:
+# <https://github.com/godotengine/godot/pull/91794>
+for key, _ in opts.UnknownVariables().items():
```

Comment:
Forgive my ignorance, but am I correct to assume that `opts.UnknownVariables().items()` is a list of command line arguments that didn't match to an existing command line option?

Reply:
Yes, it is correct if by "existing command line option" you mean something like: `opts.Add(BoolVariable("vulkan", "...", True))`.

And `.items()` is redundant (I forgot to delete it after tests :sweat_smile: ) but just formally it still works with it.

Reply:
>Yes, it is correct if by "existing command line option" you mean something like: opts.Add(BoolVariable("vulkan", "...", True)).

I meant that each platform has it's own platform specific options which added only if your target build platform is that platform. For example `linux` defines `alsa`, `pulseaudio` ... If you build for windows, providing them will be detected by `opts.UnknownVariables()` as unknown... It is a reason why I decided to create PR that specifically targets exactly `builtin_*` options as all of them are defined inside main file so won't have problems. This problem is a reason why #91213 is very good thing but needs to wait for Akien profile loading approach to be merged in one or another way.

If you define variables not in the main file, you will encounter some edge cases. For example `env["<var>"` will sometimes fail. I think it is the main reason why many of the platform specific options are defined inside main file. ([vsproj](https://github.com/godotengine/godot/pull/98470), [xaudio2](https://github.com/godotengine/godot/pull/98512), metal, d3d12...)

The problem of treating these options is a big net of interconnected features which seems independent... But all of them are connected with one thing. SCons have very weird variable handling so without Akiens approach it is possible to fix it but it will be messy. Something like 4-8 new lines of unnecessary code for each new line added with Akiens approach